composables/useFacet.ts

```diff
@@ -0,0 +1,43 @@
+import { ref, computed } from "@vue/composition-api"
+import { categoryTreeQuery } from "../gql/queries"
+import { selectCategoryFromTree } from "./helpers/selectCategoryFromTree"
+import { useState, useNuxtApp } from "#app"
+
+export const useFacet = (referenceKey: string) => {
+  const nuxt = useNuxtApp()
+  const fetcher = nuxt.nuxt2Context.$gqlFetch
+  const result = ref({})
+  const loading = useState(`use-category-loading-${referenceKey}`, () => false)
+  const error = useState(`use-category-error-${referenceKey}`, () => null)
+  const categories = ref()
+  const search = async (params: { categoryCode: string }) => {
+    try {
+      loading.value = true
+      const { categoryCode } = params
+      const categoryTree = useState(`use-category-${referenceKey}`, () => {
```

Comment:
categoryTree does not need to be a useState reference.

Reply:
Resolved